tests/unit/plugins/become/test_ksusu.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2012-2014, Michael DeHaan <michael.dehaan@gmail.com>
+# Copyright (c) 2020 Ansible Project
+#
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+
+from ansible import context
+
+from .helper import call_become_plugin
+
+
+def _test_ksu_task(mocker, parser, task, shell='/bin/bash', cmd='/bin/foo'):
+    context._init_global_context(parser.parse_args([]))
+
+    # Task arguments and defaults:
+    become_user_ksu = task.get('become_user_ksu', 'root')
+    become_user = task.get('become_user', 'root')
+    become_exe_ksu = task.get('become_exe_ksu', 'ksu')
+    become_exe_su = task.get('become_exe_su', '/bin/su')
+    become_flags_ksu = task.get('become_flags_ksu', '-Z\\s+-q')
+    become_flags = task.get('become_flags', '')
+    become_pass = task.get('become_pass', None)
+    prompt_l10n = task.get('prompt_l10n', [])
+
+    task['become_method'] = 'community.general.ksusu'  # Force the become method.
+
+    # We are appending here instead of formatting so the parts are easily understood.
+    regex = ''
+    regex += become_exe_ksu
+    regex += '\\s+' + become_user_ksu
+    regex += '\\s+' + become_flags_ksu
+    regex += '\\s+-e\\s+' + become_exe_su
+    regex += '\\s+-l\\s+' + become_flags
+    regex += '\\s+' + become_user
+    regex += '\\s+' + shell
+    regex += "\\s+-c\\s+'echo\\s+BECOME-SUCCESS-.+?\\s*;\\s*" + cmd
+    regex += "\\s*'"
```

Comment:
Following up on your Python-as-a-new-language adventure:
```python
    regex = become_exe_ksu
    regex += r'\s+' + become_user_ksu
    regex += r'\s+' + become_flags_ksu
    regex += r'\s+-e\s+' + become_exe_su
    regex += r'\s+-l\s+' + become_flags
    regex += r'\s+' + become_user
    regex += r'\s+' + shell
    regex += r"\s+-c\s+'echo\s+BECOME-SUCCESS-.+?\s*;\s*" + cmd
    regex += "\s*'"
```
But I think a more pythonic way to do this, could perhaps be:
```suggestion
    regex = (
        r"{exe_ksu}\s+{user_ksu}\s+{flags_ksu}\s+"
        r"-e\s+{exe_su}\s+-l\s+{flags}\s+{user}\s+{shell}"
        r"\s+-c\s+'echo\s+BECOME-SUCCESS-.+?\s*;\s*{cmd}\s*".format(
            exe_ksu=become_exe_ksu,
            user_ksu=become_user_ksu,
            flags_ksu=become_flags_ksu,
            exe_su=become_exe_su,
            flags=become_flags,
            user=become_user,
            shell=shell,
            cmd=cmd,
        )
    )
```